SConstruct

```diff
@@ -237,41 +269,61 @@ if 'FRAMEWORKS' not in env:
 
 add_RegressionTest(env)
 
-class defaults: pass
+defaults.prefix = {"Windows": "$ProgramFiles\Cantera", "default": "/usr/local"}
+defaults.boostIncDir = ""
 
 if os.name == 'posix':
-    defaults.prefix = '/usr/local'
-    defaults.boostIncDir = ''
     env['INSTALL_MANPAGES'] = True
 elif os.name == 'nt':
     defaults.prefix = pjoin(os.environ['ProgramFiles'], 'Cantera')
-    defaults.boostIncDir = ''
     env['INSTALL_MANPAGES'] = False
 else:
     print("Error: Unrecognized operating system '%s'" % os.name)
     sys.exit(1)
 
+defaults.CXX = "${CXX}"
+defaults.CC = "${CC}"
+
 compiler_options = [
     ('CXX',
      """The C++ compiler to use.""",
-     env['CXX']),
+     defaults.CXX),
     ('CC',
      """The C compiler to use. This is only used to compile CVODE.""",
-     env['CC'])]
+     defaults.CC)]
+
+compiler_options = substitute(compiler_options, env=env)
 opts.AddVariables(*compiler_options)
 opts.Update(env)
 
-defaults.cxxFlags = ''
-defaults.ccFlags = ''
-defaults.noOptimizeCcFlags = '-O0'
-defaults.optimizeCcFlags = '-O3'
-defaults.debugCcFlags = '-g'
-defaults.noDebugCcFlags = ''
-defaults.debugLinkFlags = ''
-defaults.noDebugLinkFlags = ''
-defaults.warningFlags = '-Wall'
-defaults.buildPch = False
-defaults.sphinx_options = '-W --keep-going'
+defaults.cxxFlags = {"cl": "/EHsc", "Cygwin": "-std=gnu++11", "default": "-std=c++11"}
+defaults.ccFlags = {
```

Comment:
If we're redoing this anyways, can you change them from camelCase to snake_case?